hasjob/views/admindash.py

```diff
@@ -4,117 +4,123 @@
-    elif q == 'c':
-        userdays = db.session.query('month', 'users', 'centile').from_statement(
-            '''SELECT month, COUNT(user_id) AS users, PERCENTILE_CONT(ARRAY [0.5, 0.6, 0.7, 0.8, 0.9, 0.91, 0.92, 0.93, 0.94, 0.95, 0.96, 0.97, 0.98, 0.99, 1.0]) WITHIN GROUP (ORDER BY count) AS centile FROM (SELECT month, COUNT(day) AS count, user_id FROM (SELECT DATE_TRUNC('month', user_active_at.active_at) AS month, DATE_TRUNC('day', user_active_at.active_at) AS day, user_active_at.user_id FROM user_active_at, job_application WHERE user_active_at.user_id = job_application.user_id AND DATE_TRUNC('month', user_active_at.active_at) = DATE_TRUNC('month', job_application.created_at) GROUP BY month, day, user_active_at.user_id) AS month_day_user GROUP BY month, user_id ORDER BY month, user_id) AS daycounts WHERE month > '2013-01-01' GROUP BY month ORDER BY month;''')
-    elif q == 'r':
-        userdays = db.session.query('month', 'users', 'centile').from_statement(
-            '''SELECT month, COUNT(user_id) AS users, PERCENTILE_CONT(ARRAY [0.5, 0.6, 0.7, 0.8, 0.9, 0.91, 0.92, 0.93, 0.94, 0.95, 0.96, 0.97, 0.98, 0.99, 1.0]) WITHIN GROUP (ORDER BY count) AS centile FROM (SELECT month, COUNT(day) AS count, user_id FROM (SELECT DATE_TRUNC('month', user_active_at.active_at) AS month, DATE_TRUNC('day', user_active_at.active_at) AS day, user_active_at.user_id FROM user_active_at, job_application WHERE user_active_at.user_id = job_application.user_id AND job_application.response = :response AND DATE_TRUNC('month', user_active_at.active_at) = DATE_TRUNC('month', job_application.replied_at) GROUP BY month, day, user_active_at.user_id) AS month_day_user GROUP BY month, user_id ORDER BY month, user_id) AS daycounts WHERE month > '2013-01-01' GROUP BY month ORDER BY month;''').params(
-            response=EMPLOYER_RESPONSE.REPLIED)
-
-    outfile = StringIO()
-    out = unicodecsv.writer(outfile, 'excel')
-    out.writerow(['month', '50%', '60%', '70%', '80%', '90%',
-        '91%', '92%', '93%', '94%', '95%', '96%', '97%', '98%', '99%', '100%', 'users'])
-
-    for month, users, centile in userdays:
-        out.writerow([month.strftime('%Y-%m')] + [round(c, 2) for c in centile] + [users])
-
-    return outfile.getvalue(), 200, {'Content-Type': 'text/plain'}
+from ..models import db, POSTSTATUS, EMPLOYER_RESPONSE
+from .admin import AdminView
+
+
+@route('/admin/dashboard')
+class AdminDashboard(AdminView):
+    __decorators__ = [lastuser.requires_permission('siteadmin')]
+
+    @route('')
+    @viewdata(tab=True, index=0, title=__("Dashboard"))
+    @lastuser.requires_permission('siteadmin')
```

Comment:
This looks like it doesn't need to be required, since it's declared at the class level.